packages/commonwealth/server/routes/bulkOffchain.ts

```diff
@@ -191,13 +194,37 @@ const bulkOffchain = async (models: DB, req: Request, res: Response) => {
         reject(new ServerError('Could not fetch contracts'));
       }
     }),
+    models.Topic.findAll({
```

Comment:
Still not SUPER happy with having the topics call here rather than part of /bulkTopics / the topics controller. However don't want to push more additional refactoring work on your plate if that's going to be a big lift -- and honestly this solution is more future-proof (if we want to do gating outside of just topics), so we can leave it for now.

Reply:
It would be a pretty big lift to transfer this logic to /bulkTopics and tbh I see advantages to having it here. `/bulkTopics` is called in many places which would create repetitive fetching of gates when we really only need and want to know them when loading up a community(also to store on the chain mode). It also helps with tokenBalance fetching(ie gates are fetched early on so we can determine if we should get token balances early and dispatch that call)

++ future non-topic gating.